packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -51,10 +108,13 @@ export function useGlobalSetting( path, blockName, source = 'all' ) {
 		} );
 	};
 
-	const getSettingValueForContext = ( name ) => {
-		const currentPath = ! name
-			? `settings.${ path }`
-			: `settings.blocks.${ name }.${ path }`;
+	const getSettingValueForContext = ( currentBlockName, settingPath ) => {
+		const currentPathParts = [ 'settings' ];
```

Comment:
This getter is going to be heavily used during re-renders, so it would be beneficial to optimize it as much as possible. One way to do this is to express the logic as:
```
const currentPath = currentBlockName ? `settings.blocks.${currentBlockName}.${settingPath}` : `settings.${settingPath}`;
```
This getter is going to be heavily used during re-renders and other performance-sensitive operations.